astropy/io/fits/time.py

```diff
@@ -0,0 +1,174 @@
+        self.time_columns = OrderedDict()
+        for i in range(1, nfields + 1):
+            self.time_columns[i] = None
+
+    def set_global_time(self, key, value, comment):
+        """
+        Set the global time reference frame attributes.
+
+        Parameters
+        ----------
+        key : string
+            FITS global time reference frame keyword.
+        value : int, str, list
+            value associated with specified keyword.
+        comment : str
+            comment associated with specified keyword.
+        """
+        if key == 'TIMESYS':
+            self.scale = value
+        elif key == 'MJDREF':
+            self.mjd = value
+        elif key == 'JDREF':
+            self.jd = value
+        elif key == 'DATEREF':
+            self.date = value
+        elif key == 'TREFPOS':
+            self.pos = value
+        elif key == 'TREFDIR':
+            self.dir = value
+        elif key == 'TIMEUNIT':
+            self.unit = value
+        elif key == 'TIMEOFFS':
+            self.offs = value
+        elif key == 'OBSGEO-X':
+            self.loc_x = value
+        elif key == 'OBSGEO-Y':
+            self.loc_y = value
+        elif key == 'OBSGEO-Z':
+            self.loc_z = value
+
+    def set_column_override(self, key, value, comment):
+        """
+        Set the time column specific override attributes.
+
+        Parameters
+        ----------
+        key : string
+            FITS time column specific keyword.
+        value : int, str, list
+            value associated with specified keyword.
+        comment : str
+            comment associated with specified keyword.
+        """
+        idx = int(key[-1])
+        if self.time_columns[idx] is None:
+            self.time_columns[idx] = OrderedDict()
+        if key[:-1] == 'TCTYP':
+            self.time_columns[idx]['scale'] = value
+        elif key[:-1] == 'TRPOS':
+            self.time_columns[idx]['pos'] = value
+        elif key[:-1] == 'TCUNI':
+            self.time_columns[idx]['unit'] = value
+        elif key[:-1] == 'TCAPF':
+            self.time_columns[idx]['APy_format'] = value
+
+    def convert_to_time(self, table):
+        """
+        Convert time columns to Astropy Time columns.
+
+        Parameters
+        ----------
+        table : astropy.table.Table
+            The table whose time columns are to be converted.
+        """
+        for idx, time_col in self.time_columns.items():
+            if time_col is not None:
+                time_colname = table.colnames[idx - 1]
+                if time_col['APy_format'] is not None:
+                    table[time_colname] = Time(table[time_colname][:,0], table[time_colname][:,1], format='jd', scale=time_col['scale'].lower())
+                    table[time_colname].format = time_col['APy_format'].lower()
+                    table[time_colname].location = EarthLocation(Quantity(self.loc_x, unit='m'), Quantity(self.loc_y, unit='m'), Quantity(self.loc_z, unit='m'))
```

Comment:
There might not be a `location` defined so you need to handle that case.  (And TEST!).